### PR TITLE
Add logic for deployment with track 2 SDK

### PR DESCRIPTION
**Pull request by Flanker32**
Add logic for deployment with track 2 SDK

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java [hunk 1]
```diff
@@ -149,193 +133,108 @@ private IWebAppDeploymentSlot createDeploymentSlot(final IWebAppDeploymentSlot s
                 .commit();
     }
 
+    private void deploy(IAppService target, WebAppConfig config) throws AzureExecutionException {
+        final DeploymentUtil deploymentUtil = new DeploymentUtil(target);
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java:137]: usually util class will not have a state

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java [hunk 2]
```diff
@@ -149,193 +133,108 @@ private IWebAppDeploymentSlot createDeploymentSlot(final IWebAppDeploymentSlot s
...
+        target.deploy(DeployType.ZIP, zipFile);
     }
 
-    private void deployExternalResources(DeployTarget deployTarget, List<Resource> externalResources) throws AzureExecutionException {
-        if (externalResources.isEmpty()) {
-            return;
-        }
-        final PublishingProfile publishingProfile = deployTarget.getPublishingProfile();
-        final String serverUrl = publishingProfile.ftpUrl().split("/", 2)[0];
+    private File prepareStagingDirectory(List<File> files) throws AzureExecutionException {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java:182]: static?

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/DeployUtils.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        // convert null to empty string
+        targetPath = StringUtils.defaultString(targetPath);
+        return StringUtils.startsWith(targetPath, "/") ? targetPath :
+                FTP_ROOT.resolve(Paths.get(targetPath)).normalize().toString();
+    }
+
+    /**
+     * Rename project jar to app.jar for java se app service
+     */
+    public static void prepareJavaSERuntime(final List<File> artifacts, final String finalName) throws AzureExecutionException {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/DeployUtils.java:74]: prepareJavaSERuntimeJar

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java [hunk 4]
```diff
@@ -149,193 +133,108 @@ private IWebAppDeploymentSlot createDeploymentSlot(final IWebAppDeploymentSlot s
                 .commit();
     }
 
+    private void deploy(IAppService target, WebAppConfig config) throws AzureExecutionException {
+        final DeploymentUtil deploymentUtil = new DeploymentUtil(target);
+        try {
+            deploymentUtil.beforeDeployArtifacts();
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java:139]: DeploymentUtil.beforeDeployArtifacts(target);

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/DeployUtils.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.jar.JarInputStream;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+
+import static com.microsoft.azure.maven.webapp.handlers.artifact.ArtifactHandlerUtils.DEFAULT_APP_SERVICE_JAR_NAME;
+
+public class DeployUtils {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/DeployUtils.java:29]: DeploymentUtil DeployUtils confusing

Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/utils/DeployUtils.java:29]: Deleted `DeploymentUtil`, move the start/stop method to WebAppUtils

**Review by andxu — APPROVED** [review]

**Pull request merged**